pkg/provider/cloud/kubevirt/client.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubevirt
+
+import (
+	"encoding/base64"
+
+	kv "k8c.io/kubermatic/v2/pkg/provider/cloud/kubevirt/kubevirtcli/client/versioned"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+type Clientset struct {
+	RestConfig *restclient.Config
+	// KubeVirt clientset
+	KubeVirt kv.Interface
+	// Kubernetes clientset
+	Kubernetes kubernetes.Interface
+}
+
+func NewClientset(kubeconfig string) (*Clientset, error) {
+	config, err := base64.StdEncoding.DecodeString(kubeconfig)
+	if err != nil {
+		// if the decoding failed, the kubeconfig is sent already decoded without the need of decoding it,
```

Comment:
So the problem with that is, in the past we had some issues passing and parsing the Kubeconfig as a json or yaml from UI to API and vice versa. Thus we demanded that this should be base64 encoded so we avoid all the issues that comes a long parsing data as plaintext